codalab/lib/bundle_cli.py

```diff
@@ -1718,6 +1719,30 @@ def do_make_command(self, args):
             params={'worksheet': worksheet_uuid},
         )
 
+        # Add new bundle's location. If user specify the storage using `--store`, the bundle will be added to that storage.
+        # Otherwise, the new MakeBundle will be added to default storage, which is set by the rest server.
+        destination_bundle_store = metadata.get('store')
+        if destination_bundle_store is not None:
+            need_bypass = False
+            bundle_store_uuid = None
+
+            # 1) Read destination store from --store if user has specified it
+            if destination_bundle_store is not None and destination_bundle_store != '':
+                storage_info = client.fetch_one(
+                    'bundle_stores',
+                    params={
+                        'name': destination_bundle_store,
+                        'include': ['uuid', 'storage_type', 'url'],
+                    },
+                )
+                bundle_store_uuid = storage_info['uuid']
+                if storage_info['storage_type'] in (StorageType.DISK_STORAGE.value,):
+                    need_bypass = False  # The user specify --store to upload to disk storage
+
+            # 2) Add bundle to destination bundle storage
+            params = {'need_bypass': need_bypass}  # is_dir: ???
+            client.add_bundle_location(new_bundle['uuid'], bundle_store_uuid, params)
+
```

Comment:
Should we add code that handles the case when `destination_bundle_store` is not specified (and needs to be set to a default value)?

Reply:
Great question! The client.add_bundle_locations function will trigger the client side to handle this. This means if the user does not specify the `destination_bundle_store`, rest server will add the new bundle to default storage